compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIEUtils.h

```diff
@@ -135,6 +135,10 @@ std::string getConstantIntValuesString(ArrayRef<OpFoldResult> opFoldResults);
 bool sinkInto(Region &, IRRewriter &,
               std::function<bool(Operation *)> shouldSink);
 
+/// Annotate `forOp` with the llvm.loop_annotation attribute specifying that
+/// it should never be unrolled.
+void addNoUnrollAttribute(scf::ForOp forOp, OpBuilder &builder);
```

Comment:
What should happen if the loop already has a 'NoUnroll' loop annotation? And unless I overlooked, you don't have a testcase for that?